usecases/ingestion_usecase.go

```diff
@@ -280,14 +281,16 @@ func (usecase *IngestionUseCase) processUploadLog(ctx context.Context, uploadLog
 		}
 	}
 
-	file, err := usecase.gcsRepository.GetFile(ctx, usecase.GcsIngestionBucket, uploadLog.FileName)
+	file, err := usecase.blobRepository.GetBlob(ctx, usecase.GcsIngestionBucket, uploadLog.FileName)
+	if file.ReadCloser != nil {
+		defer file.ReadCloser.Close()
+	}
```

Comment:
Ah, yes I did them in both orders.
Honestly it's equivalent, writer is expected to be nil if err is not, and vice versa.
The important thing with writers is to:
- especially close them when you are done writing and check the err response, because it may still finish writing stuff from the previous call to Write (there is a buffer)
- preferably still defer Close() it to be sure you don't leave dangling writers hanging